app/actions/local/notification.ts

```diff
@@ -16,13 +13,13 @@ export const updatePostSinceCache = async (serverUrl: string, notification: Noti
     try {
         if (notification.payload?.channel_id) {
             const {database} = operator;
-            const chunks = await queryPostsInChannel(database, notification.payload.channel_id);
+            const chunks = await queryPostsInChannel(database, notification.payload.channel_id).fetch();
             if (chunks.length) {
                 const recent = chunks[0];
-                const lastPost = await database.get<PostModel>(MM_TABLES.SERVER.POST).find(notification.payload.post_id);
+                const lastPost = await getPostById(database, notification.payload.post_id);
                 await operator.database.write(async () => {
                     await recent.update(() => {
-                        recent.latest = lastPost.createAt;
+                        recent.latest = lastPost!.createAt;
```

Comment:
how are we sure `lastPost` is not undefined? I rather wrap the hole write operation with an `if (lastPost)`

Reply:
The theory is that if the post is in the POST_IN_CHANNEL table, it should be in the post table. But if there is any data inconsistency, you are right, this may be undefined. Let's do it.